test/e2e/util/util.go

```diff
@@ -216,6 +214,92 @@ func WaitForNUnReadyNodes(t *testing.T, ctx context.Context, client crclient.Cli
 	return nodes.Items
 }
 
+func WaitForNReadyNodesByNodePool(t *testing.T, ctx context.Context, client crclient.Client, n int32, platform hyperv1.PlatformType, nodePoolName string) []corev1.Node {
+	g := NewWithT(t)
+	start := time.Now()
+
+	// waitTimeout for nodes to become Ready
+	waitTimeout := 30 * time.Minute
+	switch platform {
+	case hyperv1.PowerVSPlatform:
+		waitTimeout = 60 * time.Minute
+	}
+
+	t.Logf("Waiting for nodes to become ready by NodePool. NodePool: %s Want: %v", nodePoolName, n)
+	nodes := &corev1.NodeList{}
+	nodesFromNodePool := []corev1.Node{}
+	readyNodeCount := 0
```

Comment:
nit: `nodesFromNodePool`, `readyNodeCount`, and `readyNodes` seem redundant. Seems like you could have everything you need with `nodesFromNodePool`